data/lua/core.lua

```diff
@@ -103,8 +105,8 @@ end
 
 local create_tag_mt = {
 	__metatable = "WML tag builder",
-	__index = function(self, n)
-		return function(cfg) return { n, cfg } end
+	__index = function(_, tag_name)
```

Comment:
I'm not really a fan of using `_` for unused variables. I don't mind it so much for loop variables, but I wouldn't do it for function parameters. _Especially_ not when the parameter was named `self` (or is otherwise intended to be the implicit parameter with the `obj:func()` notation).

Reply:
Let's move discussion to one place: https://github.com/wesnoth/wesnoth/pull/2380#discussion_r162595394